EssentialFeed/EssentialFeedTests/Image Comments/API/ImageCommentsMapperTests.swift

```diff
@@ -0,0 +1,83 @@
+//
+// Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import XCTest
+import EssentialFeed
+
+class ImageCommentsMapperTests: XCTestCase {
+	func test_map_throwsErrorOnNon2xxHTTPResponse() throws {
+		let json = makeItemsJSON([])
+		let samples = [199, 150, 300, 400, 500]
+
+		try samples.forEach { code in
+			XCTAssertThrowsError(
+				try ImageCommentsMapper.map(json, from: HTTPURLResponse(statusCode: code))
+			)
+		}
+	}
+
+	func test_map_throwsErrorOn2xxHTTPResponseWithInvalidJSON() throws {
+		let invaliJSON = Data("invalid json".utf8)
```

Comment:
Typographical error:
```suggestion
		let invalidJSON = Data("invalid json".utf8)
```